pkg/system/system.go

```diff
@@ -345,10 +346,11 @@ func (sc *Controller) upgradeDaemons() func(w http.ResponseWriter, r *http.Reque
 				}
 			}
 
-			// TODO: why renaming?
-			err = os.Rename(c.NydusdPath, manager.NydusdBinaryPath)
+			cmd := exec.Command("sudo", "install", "-D", "-m", "755", c.NydusdPath, manager.NydusdBinaryPath)
```

Comment:
Thanks for the fixup! Can we use `io.Copy` and `os.Chmod` instead of shell command here? Not all envs have sudo and install commands.

Reply:
done.